app/src/main/java/com/rage/clamber/Fragments/ClimbDetailFragment.java

```diff
@@ -0,0 +1,373 @@
+        if (networkInfo != null && networkInfo.isConnected()) {
+
+            Call<Double> avgRatingCall = ApiManager.getClamberService().getUserAverageRatingForClimb(climbId);
+            avgRatingCall.enqueue(new Callback<Double>() {
+                @Override
+                public void onResponse(Call<Double> call, Response<Double> response) {
+                    if (response.code() == 200) {
+                        userAvgRating = response.body();
+                        userRatingTextView.setVisibility(View.VISIBLE);
+                        if (userAvgRating == NO_DATA) {
+                            userRatingTextView.setText(getContext().getString(R.string.user_rating_s, ""));
+                        } else {
+                            userRatingTextView.setText(getContext().getString(R.string.user_rating_s, userAvgRating));
+                        }
+                    } else {
+                        Log.d(TAG, "Non 200 response code returned");
+                    }
+                }
+
+                @Override
+                public void onFailure(Call<Double> call, Throwable t) {
+                    Log.d(TAG, "Failure when attempting to get average ratings");
+                }
+            });
+
+        } else {
+            Toast.makeText(getContext(), R.string.no_internet_connection, Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    /**
+     * Method to add a rating to the database for a user based off of up and down arrow button clicks.
+     * @param request - request contains the climb id, user name and rating. Information is populated
+     *                in the onClick methods for the up and down arrows.
+     */
+    public void addRatingForUser(UserRatingRequest request){
+        final Call<Integer> addRatingCall = ApiManager.getClamberService().addUserRating(request);
+        addRatingCall.enqueue(new Callback<Integer>() {
+            @Override
+            public void onResponse(Call<Integer> call, Response<Integer> response) {
+                if (response.code() == 200) {
+                    yourRatingTextView.setText(getContext().getString(R.string.your_rating_s, response.body()));
+                    getAvgUserRatingForClimb(climb.getClimbId());
+                    userRatingTextView.setText(getContext().getString(R.string.user_rating_s, userAvgRating));
+                }
+                else {
+                    Log.d(TAG, "Non 200 response code returned - check server");
+                }
+            }
+
+            @Override
+            public void onFailure(Call<Integer> call, Throwable t) {
+                Log.d(TAG, "Failed to add rating for user");
+            }
+        });
+    }
+
+
+    /**
+     * When the Up Arrow Button is selected, the add rating method is called and given the
+     * climb id, user name and the existing rating plus one.
+     */
+    @OnClick(R.id.comments_fragment_up_arrow)
```

Comment:
you could create another function like, `onArrowClicked` that takes an argument specifying whether to go up or down, then both `onUpArrowClicked` and `onDownArrowClicked` could just call that. That way you could avoid duplicating the logic in those two functions, since they're the same apart from +1/-1.